dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/InferredEntriesHelper.java

```diff
@@ -0,0 +1,221 @@
+package io.dockstore.webservice.helpers;
+
+import io.dockstore.common.DescriptorLanguage;
+import io.dockstore.common.EntryType;
+import java.io.StringWriter;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+public class InferredEntriesHelper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(InferredEntriesHelper.class);
+
+    public List<InferredEntry> infer(FileTree fileTree) {
+
+        // Get a list of all paths in the repo.
+        List<String> allPaths = fileTree.listPaths();
```

Comment:
Note for understanding, presumably this is one call (albeit a large one with seems to have protections for spamming)

Reply:
Yup, the underlying GitHub API call is
https://docs.github.com/en/rest/git/trees?apiVersion=2022-11-28#get-a-tree